src/main/java/com/kodilla/ecommercee/service/GroupService.java

```diff
@@ -0,0 +1,40 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.Group;
+import com.kodilla.ecommercee.exception.GroupNotFoundException;
+import com.kodilla.ecommercee.repository.GroupRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class GroupService {
+
+    private final GroupRepository groupRepository;
+
+    public void createGroup(Group group) {
+        groupRepository.save(group);
+    }
+
+    public List<Group> getAllGroups() {
+        return groupRepository.findAll();
+    }
+
+    public Group getGroup(final Long id) throws GroupNotFoundException {
```

Comment:
`throws GroupNotFoundException` is redundant - you make exception here `orElseThrow(GroupNotFoundException::new)`